tests/Integrations/PDO/PDOSandboxedTest.php

```diff
@@ -0,0 +1,398 @@
+            SpanAssertion::build('PDO.prepare', 'PDO', 'sql', "WRONG QUERY")
+                ->withExactTags($this->baseTags()),
+            SpanAssertion::exists('PDO.__construct'),
+        ]);
+    }
+
+    public function testPDOStatementException()
+    {
+        $query = "WRONG QUERY";
+        $traces = $this->isolateTracer(function () use ($query) {
+            try {
+                $pdo = $this->pdoInstance();
+                $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
+                $stmt = $pdo->prepare($query);
+                $stmt->execute([1]);
+                $stmt->fetchAll();
+                $stmt->closeCursor();
+                $stmt = null;
+                $pdo = null;
+            } catch (\PDOException $ex) {
+            }
+        });
+        $this->assertSpans($traces, [
+            SpanAssertion::build('PDOStatement.execute', 'PDO', 'sql', $query)
+                ->setTraceAnalyticsCandidate()
+                // phpcs:disable
+                ->setException('PDOException', 'SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near \'WRONG QUERY\' at line 1')
+                // phpcs:enable
+                ->withExactTags($this->baseTags()),
+            SpanAssertion::build('PDO.prepare', 'PDO', 'sql', $query)
+                ->withExactTags($this->baseTags()),
+            SpanAssertion::exists('PDO.__construct'),
+        ]);
+    }
+
+    // Add limited tracer to sandboxed closures
+    /*
+    public function testLimitedTracerPDO()
+    {
+        $query = "SELECT * FROM tests WHERE id = ?";
+        $traces = $this->isolateLimitedTracer(function () use ($query) {
+            $pdo = $this->pdoInstance();
+            $stmt = $pdo->prepare($query);
+            $stmt->execute([1]);
+            $results = $stmt->fetchAll();
+            $this->assertEquals('Tom', $results[0]['name']);
+            $stmt->closeCursor();
+            $stmt = null;
+            $pdo = null;
+        });
+
+        $this->assertEmpty($traces);
+    }
+    */
+
+    // @see https://github.com/DataDog/dd-trace-php/issues/510
+    public function testPDOStatementsAreReleased()
+    {
+        $query = "SELECT * FROM tests WHERE id = ?";
+        $traces = $this->isolateTracer(function () use ($query) {
+            $pdo = new \PDO(
+                $this->mysqlDns(),
+                self::MYSQL_USER,
+                self::MYSQL_PASSWORD,
+                [\PDO::ATTR_EMULATE_PREPARES => false]
+            );
+            $stmt = $pdo->prepare($query);
+            $stmt->execute([1]);
+            unset($stmt);
+
+            $pdo->prepare($query)->execute([2]);
+
+            $closedCount = $pdo->query("SHOW SESSION STATUS LIKE 'Com_stmt_close'")->fetchColumn(1);
+
+            $this->assertEquals(2, $closedCount);
+        });
+    }
+
```

Comment:
Some new tests are missing from this file:
https://github.com/DataDog/dd-trace-php/blob/master/tests/Integrations/PDO/PDOTest.php#L242
and
https://github.com/DataDog/dd-trace-php/blob/master/tests/Integrations/PDO/PDOTest.php#L258